tests/v2/validation/airgap/rke1_custom_cluster_test.go

```diff
@@ -0,0 +1,147 @@
+}
+
+func (a *AirGapRKE1CustomClusterTestSuite) TearDownSuite() {
+	a.session.Cleanup()
+}
+
+func (a *AirGapRKE1CustomClusterTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	a.session = testSession
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	corralRancherHA := new(corralha.CorralRancherHA)
+	config.LoadConfig(corralha.CorralRancherHAConfigConfigurationFileKey, corralRancherHA)
+
+	registriesConfig := new(registries.Registries)
+	config.LoadConfig(registries.RegistriesConfigKey, registriesConfig)
+
+	a.kubernetesVersions = clustersConfig.RKE1KubernetesVersions
+	a.cnis = clustersConfig.CNIs
+	a.advancedOptions = clustersConfig.AdvancedOptions
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(a.T(), err)
+
+	a.client = client
+	listOfCorrals, err := corral.ListCorral()
+	require.NoError(a.T(), err)
+
+	corralConfig := corral.CorralConfigurations()
+
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(a.T(), err)
+
+	corralPackage := corral.CorralPackagesConfig()
+	a.corralImage = corralPackage.CorralPackageImages[corralPackageAirgapCustomClusterName]
+	a.corralAutoCleanup = corralPackage.Cleanup
+
+	_, corralExist := listOfCorrals[corralRancherHA.Name]
+	if corralExist {
+		bastionIP, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryIP)
+		require.NoError(a.T(), err)
+
+		err = corral.UpdateCorralConfig(corralBastionIP, bastionIP)
+		require.NoError(a.T(), err)
+
+		registryFQDN, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryFQDN)
+		require.NoError(a.T(), err)
+		logrus.Infof("registry fqdn is %s", registryFQDN)
+		a.registryFQDN = registryFQDN
+	} else {
+		a.registryFQDN = registriesConfig.ExistingNoAuthRegistryURL
+	}
+
+}
+
+func (a *AirGapRKE1CustomClusterTestSuite) TestProvisioningRKE1CustomCluster() {
+	nodeRoles := map[int]string{
+		1: "--etcd --controlplane --worker",
+	}
+
+	var name string
+	for _, kubeVersion := range a.kubernetesVersions {
+		name = "Airgap Provision RKE1 Custom Cluser Kubernetes version: " + kubeVersion
+		for _, cni := range a.cnis {
+			name += " cni: " + cni
+			a.Run(name, func() {
+				clusterName := testProvisioningRKE1CustomCluster(a.T(), a.client, nodeRoles, a.corralImage, cni, kubeVersion, a.registryFQDN, a.corralAutoCleanup, a.advancedOptions)
+				checkPodStatusImageSource(a.T(), a.client, clusterName, a.registryFQDN)
+			})
+		}
+	}
+
+}
+
+func (a *AirGapRKE1CustomClusterTestSuite) TestProvisioningUpgradeRKE1CustomCluster() {
+	nodeRoles := map[int]string{
+		1: "--etcd --controlplane --worker",
+	}
+
+	rke2Versions, err := kubernetesversions.ListRKE1AllVersions(a.client)
+	require.NoError(a.T(), err)
+
+	numOfRKE2Versions := len(rke2Versions)
+	// for this we will only have one custom cluster entry and one cni entry
```

Comment:
same nit as above; is this comment needed?

Reply:
Same as above, but I can remove if you think it's unnecessary.